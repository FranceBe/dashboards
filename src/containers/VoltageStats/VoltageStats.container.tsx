import ChartBar from 'components/ChartBar'
import { VoltageStatContainer } from 'containers/VoltageStats/voltageStats.style'
import React from 'react'
import { VoltageStatsProps } from 'types/voltageStats'
export const VoltageStats: React.FC<VoltageStatsProps> = ({ devicesVoltage }) => {
  return (
    <VoltageStatContainer>
      <h2>Voltage per device</h2>
      <ChartBar data={devicesVoltage} valueKey={'voltage'} nameKey={'name'} />
    </VoltageStatContainer>
  )
}

export default VoltageStats
