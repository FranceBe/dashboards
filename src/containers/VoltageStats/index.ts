import VoltageStatsContainer, { VoltageStats } from 'containers/VoltageStats/VoltageStats.container'

export { VoltageStats }

export default VoltageStatsContainer
