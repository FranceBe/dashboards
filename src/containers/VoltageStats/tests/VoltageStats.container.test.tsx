import '@testing-library/jest-dom/extend-expect'

import { render, screen } from '@testing-library/react'
import VoltageStats from 'containers/VoltageStats'
import React from 'react'

describe('VoltageStats', () => {
  const devicesVoltage = [
    { name: 'device_0', voltage: 11.3 },
    { name: 'device_1', voltage: 45.9833 },
    { name: 'device_2', voltage: 5 },
  ]
  it('should match snapshot', () => {
    const { container } = render(<VoltageStats devicesVoltage={devicesVoltage} />)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render a ChartBar component', () => {
    render(<VoltageStats devicesVoltage={devicesVoltage} />)
    expect(screen.getByTestId('chart-bar')).toBeInTheDocument()
  })
})
