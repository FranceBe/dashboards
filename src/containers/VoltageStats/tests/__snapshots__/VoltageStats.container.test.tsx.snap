// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoltageStats should match snapshot 1`] = `
<div
  class="sc-bdnxRM bsccyY"
>
  <h2>
    Voltage per device
  </h2>
  <div
    data-testid="chart-bar"
  >
    <div
      class="recharts-wrapper"
      style="position: relative; cursor: default; width: 500px; height: 300px;"
    >
      <svg
        class="recharts-surface"
        height="300"
        version="1.1"
        viewBox="0 0 500 300"
        width="500"
      >
        <defs>
          <clippath
            id="recharts1-clip"
          >
            <rect
              height="260"
              width="430"
              x="65"
              y="5"
            />
          </clippath>
        </defs>
        <g
          class="recharts-cartesian-grid"
        >
          <g
            class="recharts-cartesian-grid-horizontal"
          >
            <line
              fill="none"
              height="260"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="430"
              x="65"
              x1="65"
              x2="495"
              y="5"
              y1="265"
              y2="265"
            />
            <line
              fill="none"
              height="260"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="430"
              x="65"
              x1="65"
              x2="495"
              y="5"
              y1="200"
              y2="200"
            />
            <line
              fill="none"
              height="260"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="430"
              x="65"
              x1="65"
              x2="495"
              y="5"
              y1="135"
              y2="135"
            />
            <line
              fill="none"
              height="260"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="430"
              x="65"
              x1="65"
              x2="495"
              y="5"
              y1="70"
              y2="70"
            />
            <line
              fill="none"
              height="260"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="430"
              x="65"
              x1="65"
              x2="495"
              y="5"
              y1="5"
              y2="5"
            />
          </g>
          <g
            class="recharts-cartesian-grid-vertical"
          >
            <line
              fill="none"
              height="260"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="430"
              x="65"
              x1="136.66666666666669"
              x2="136.66666666666669"
              y="5"
              y1="5"
              y2="265"
            />
            <line
              fill="none"
              height="260"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="430"
              x="65"
              x1="280"
              x2="280"
              y="5"
              y1="5"
              y2="265"
            />
            <line
              fill="none"
              height="260"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="430"
              x="65"
              x1="423.33333333333337"
              x2="423.33333333333337"
              y="5"
              y1="5"
              y2="265"
            />
            <line
              fill="none"
              height="260"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="430"
              x="65"
              x1="65"
              x2="65"
              y="5"
              y1="5"
              y2="265"
            />
            <line
              fill="none"
              height="260"
              offset="[object Object]"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="430"
              x="65"
              x1="495"
              x2="495"
              y="5"
              y1="5"
              y2="265"
            />
          </g>
        </g>
        <g
          class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
        >
          <line
            class="recharts-cartesian-axis-line"
            fill="none"
            height="30"
            orientation="bottom"
            stroke="#666"
            type="category"
            width="430"
            x="65"
            x1="65"
            x2="495"
            y="265"
            y1="265"
            y2="265"
          />
          <g
            class="recharts-cartesian-axis-ticks"
          >
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                orientation="bottom"
                stroke="#666"
                type="category"
                width="430"
                x="65"
                x1="136.66666666666669"
                x2="136.66666666666669"
                y="265"
                y1="271"
                y2="265"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                orientation="bottom"
                stroke="none"
                text-anchor="middle"
                type="category"
                width="430"
                x="136.66666666666669"
                y="273"
              >
                <tspan
                  dy="0.71em"
                  x="136.66666666666669"
                >
                  device_0
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                orientation="bottom"
                stroke="#666"
                type="category"
                width="430"
                x="65"
                x1="280"
                x2="280"
                y="265"
                y1="271"
                y2="265"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                orientation="bottom"
                stroke="none"
                text-anchor="middle"
                type="category"
                width="430"
                x="280"
                y="273"
              >
                <tspan
                  dy="0.71em"
                  x="280"
                >
                  device_1
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                orientation="bottom"
                stroke="#666"
                type="category"
                width="430"
                x="65"
                x1="423.33333333333337"
                x2="423.33333333333337"
                y="265"
                y1="271"
                y2="265"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                orientation="bottom"
                stroke="none"
                text-anchor="middle"
                type="category"
                width="430"
                x="423.33333333333337"
                y="273"
              >
                <tspan
                  dy="0.71em"
                  x="423.33333333333337"
                >
                  device_2
                </tspan>
              </text>
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
        >
          <line
            class="recharts-cartesian-axis-line"
            fill="none"
            height="260"
            orientation="left"
            stroke="#666"
            type="number"
            width="60"
            x="5"
            x1="65"
            x2="65"
            y="5"
            y1="5"
            y2="265"
          />
          <g
            class="recharts-cartesian-axis-ticks"
          >
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="260"
                orientation="left"
                stroke="#666"
                type="number"
                width="60"
                x="5"
                x1="59"
                x2="65"
                y="5"
                y1="265"
                y2="265"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="260"
                orientation="left"
                stroke="none"
                text-anchor="end"
                type="number"
                width="60"
                x="57"
                y="265"
              >
                <tspan
                  dy="0.355em"
                  x="57"
                >
                  0
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="260"
                orientation="left"
                stroke="#666"
                type="number"
                width="60"
                x="5"
                x1="59"
                x2="65"
                y="5"
                y1="200"
                y2="200"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="260"
                orientation="left"
                stroke="none"
                text-anchor="end"
                type="number"
                width="60"
                x="57"
                y="200"
              >
                <tspan
                  dy="0.355em"
                  x="57"
                >
                  15
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="260"
                orientation="left"
                stroke="#666"
                type="number"
                width="60"
                x="5"
                x1="59"
                x2="65"
                y="5"
                y1="135"
                y2="135"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="260"
                orientation="left"
                stroke="none"
                text-anchor="end"
                type="number"
                width="60"
                x="57"
                y="135"
              >
                <tspan
                  dy="0.355em"
                  x="57"
                >
                  30
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="260"
                orientation="left"
                stroke="#666"
                type="number"
                width="60"
                x="5"
                x1="59"
                x2="65"
                y="5"
                y1="70"
                y2="70"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="260"
                orientation="left"
                stroke="none"
                text-anchor="end"
                type="number"
                width="60"
                x="57"
                y="70"
              >
                <tspan
                  dy="0.355em"
                  x="57"
                >
                  45
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="260"
                orientation="left"
                stroke="#666"
                type="number"
                width="60"
                x="5"
                x1="59"
                x2="65"
                y="5"
                y1="5"
                y2="5"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="260"
                orientation="left"
                stroke="none"
                text-anchor="end"
                type="number"
                width="60"
                x="57"
                y="5"
              >
                <tspan
                  dy="0.355em"
                  x="57"
                >
                  60
                </tspan>
              </text>
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-bar"
        >
          <g
            class="recharts-layer recharts-bar-rectangles"
          >
            <g
              class="recharts-layer"
            >
              <g
                class="recharts-layer recharts-bar-rectangle"
              />
              <g
                class="recharts-layer recharts-bar-rectangle"
              />
              <g
                class="recharts-layer recharts-bar-rectangle"
              />
            </g>
          </g>
        </g>
      </svg>
      <div
        class="recharts-legend-wrapper"
        style="position: absolute; width: 490px; height: auto; left: 5px; top: 5px;"
      >
        <ul
          class="recharts-default-legend"
          style="padding: 0px; margin: 0px; text-align: center;"
        >
          <li
            class="recharts-legend-item legend-item-0"
            style="display: inline-block; margin-right: 10px;"
          >
            <svg
              class="recharts-surface"
              height="14"
              style="display: inline-block; vertical-align: middle; margin-right: 4px;"
              version="1.1"
              viewBox="0 0 32 32"
              width="14"
            >
              <path
                class="recharts-legend-icon"
                d="M0,4h32v24h-32z"
                fill="#0E074B"
                stroke="none"
              />
            </svg>
            <span
              class="recharts-legend-item-text"
              style="color: rgb(14, 7, 75);"
            >
              voltage
            </span>
          </li>
        </ul>
      </div>
      <div
        class="recharts-tooltip-wrapper"
        style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
      >
        <div
          class="recharts-default-tooltip"
          style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
        >
          <p
            class="recharts-tooltip-label"
            style="margin: 0px;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
